enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,593 @@
+    spec:
+      addOnConfiguration:
+        configGVR:
+          group: submarineraddon.open-cluster-management.io
+          version: v1alpha1
+          resource: SubmarinerConfig
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: SubmarinerConfig
+    metadata:
+      name: submariner
+      namespace: cluster1
+    spec:
+      IPSecIKEPort: 500
+      IPSecNATTPort: 4500
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster1
+    spec:
+      config:
+        name: submariner
+        namespace: cluster1
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: SubmarinerConfig
+    metadata:
+      name: submariner
+      namespace: cluster2
+    spec:
+      IPSecIKEPort: 501
+      IPSecNATTPort: 4501
+    
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster2
+    spec:
+      config:
+        name: submariner
+        namespace: cluster2
+    ```
+
+    then the configuration reference in this add-on status will be
+
+    ```yaml
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster1
+    spec:
+      config:
+        name: submariner
+        namespace: cluster1
+    status:
+      configReference:
+        configGVR:
+          group: proxy.open-cluster-management.io
+          version: v1alpha1
+          resource: SubmarinerConfig
+        config:
+          name: submariner
+          namesapce: cluster1
+        lastObservedGeneration: 1
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster2
+    spec:
+      config:
+        name: submariner
+        namespace: cluster2
+    status:
+      configReference:
+        configGVR:
+          group: proxy.open-cluster-management.io
+          version: v1alpha1
+          resource: SubmarinerConfig
+        config:
+          name: submariner
+          namesapce: cluster2
+        lastObservedGeneration: 1
+    ```
+
+### Test Plan
+- All implementations will have thorough coverage of unit tests.
+- Integration tests will cover the user cases
+
+### Graduation Criteria
+1. The new APIs is reviewed and accepted;
+2. Implementation is completed to support the functionalities;
+3. Develop test cases to demonstrate this proposal work correctly;
+
+### Upgrade Strategy
+- We will deprecate the elder fields of `ConfigCoordinates` in `ClusterManagementAddOn`, the add-on developers need
+adopt the new API if they used the `ConfigCoordinates` in `ClusterManagementAddOn`
+- We will deprecate the values annotation for `ManagedClusterAddOn`, the add-on developers need adopt the new API if
+they used the values annotation to configure their add-ons
```

Comment:
Is it necessary to deprecate the values annotation? Or can it be left to individual add-ons to determine how to get configuration?

I like the approach in this proposal. I like that add-ons can choose to implement their own CRD for configuration, or they can use the proposed `AddOnDeployementConfig`, or they could even just use a ConfigMap. That flexibility is very nice.

But I think that *requiring* a separate configuration resource in all cases is unnecessary for all configuration - annotations on the ManagedClusterAddon provide a simple way for basic configuration to be applied. For example, the policy addon currently supports a `log-level` annotation which I think is a nice user experience.